### ISSUE TITLE
Support CI/CD visualization configuration

### ISSUE THREAD

**Issue by Jengro777** [go-gitea/gitea#26037]
### Feature Description

The writing of yaml is cumbersome and prone to errors, and it would be great if gitea could be configured in a UI capable manner.

The following are relevant tools that can be referenced:

https://github.com/milanm/DevOps-Roadmap

**Comment by silverwind** [comment]
I don't think it's feasible given our resources. The YAML schema for actions is far too massive. Maybe if GitHub implements a reusable JS library for it.

**Comment by uvulpos** [comment]
@silverwind agree, but what we could do is we could just make a dummy visualization of the given yaml. Apparently it does help some people to have such an overview.

<img width="616" alt="Screenshot 2023-10-09 at 21 53 25" src="https://github.com/go-gitea/gitea/assets/53957681/f1df7f9e-3945-4958-a45c-c9e149bdd4a6">

**Comment by yp05327** [comment]
I agree: maybe we can implement it if GitHub implements a reusable JS library for it.
You can use other tools or IDE extensions to avoid the format errors.

**Comment by bircni** [comment]
Would definitely reject that at that point - if you still see it in another way reopen
